src/sentry/db/mixin.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import
+
+import logging
+import six
+
+from time import time
+
+from sentry.models import OrganizationOption
+
+logger = logging.getLogger('sentry.deletions')
+
+
+def delete_pending_deletion_option(instance, **kwargs):
+    if hasattr(instance, 'delete_pending_deletion_option'):
+        instance.delete_pending_deletion_option()
+
+
+class PendingDeletionMixin(object):
+    _rename_fields_on_pending_delete = []
+
+    def build_pending_deletion_key(self):
+        return 'pending-delete:%s:%s' % (self.__class__.__name__, self.id)
+
+    def rename_on_pending_deletion(self, fields=None):
+        fields = fields or self._rename_fields_on_pending_delete
+        original_data = {}
+
+        for field in fields:
+            original_data[field] = getattr(self, field)
+            setattr(self, field, '%s %d' % (original_data[field], time()))
```

Comment:
Lemme confirm if I'm understanding this correctly.

This is going to rename a field that we deem should be replaced. In what you have so far, it's `name` and `external_id` for a `Repository`. You're doing this to avoid duplicates and allow reclaiming the name.

But what'll happen is say, the `name` is `foo`, you'll end up rewriting it to `foo 1549661903`. This means we're adding 11 additional characters to the length of the field. This additional 11 characters is likely to push it over the field's max length.

This means the .save() below on L32 is going to fail.

Is there a reason we need to rewrite it like this? My proposal would be just something like:  `'!%s' % uuid4().hex`

This guarantees we're shorter than max lengths and prefixes with a ! to make it impossible to conflict with any input.

Is there any reason we were trying to preserve the original value here?

Reply:
Oh, and I now see previous discussion around this. Why do we show this in the UI when something is pending deletion? Is it so it can be cancelled? I know in most cases, we don't surface this information, and in theory, if we rename and hide it immediately, it doesn't matter?

Reply:
@mattrobenolt with Repositories we do show it in the ui. And yes it is to cancel deletion if the owner changes its mind. We don't hide it immediately here. And @markstory was concerned that users would find this confusing in that case.

Reply:
Let me see if I can handle the case of repositories specially, since it is a special case.  and for the general case I'll go back to the uuid.